File: context_os_runtime/authority.py
from __future__ import annotations

import subprocess
from pathlib import Path


def runtime_repo_root() -> Path:
    return Path(__file__).resolve().parent.parent


def verify_runtime_bundle() -> None:
    repo_root = runtime_repo_root()
    result = subprocess.run(
        ["python3", "scripts/verify_agent_os_bundle.py"],
        cwd=repo_root,
        capture_output=True,
        text=True,
        check=False,
    )
    if result.returncode != 0:
        detail = result.stdout.strip() or result.stderr.strip() or "runtime bundle verification failed"
        raise RuntimeError(detail)

File: src/core/authority.ts
import { execFile } from 'node:child_process';
import { dirname, resolve } from 'node:path';
import { fileURLToPath } from 'node:url';
import { promisify } from 'node:util';

const exec = promisify(execFile);

export function runtimeRepoRoot(): string {
  const here = dirname(fileURLToPath(import.meta.url));
  return resolve(here, '..', '..');
}

export async function verifyRuntimeBundle(): Promise<void> {
  const repoRoot = runtimeRepoRoot();
  try {
    await exec('python3', ['scripts/verify_agent_os_bundle.py'], { cwd: repoRoot });
  } catch (e) {
    const err = e as { stdout?: string; stderr?: string; message: string };
    const detail = err.stdout?.trim() || err.stderr?.trim() || 'runtime bundle verification failed';
    throw new Error(detail);
  }
}
